webui/react/src/pages/ProjectDetails.tsx

```diff
@@ -162,42 +164,58 @@ const ProjectDetails: React.FC = () => {
       const states = (settings.state || []).map((state) => (
         encodeExperimentState(state as RunState)
       ));
-      const response = await getExperiments(
+      const experimentFilter: GetExperimentsParams = {
+        archived: settings.archived ? undefined : false,
+        labels: settings.label,
+        name: settings.search,
+        orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+        projectId: id,
+        sortBy: validateDetApiEnum(V1GetExperimentsRequestSortBy, settings.sortKey),
+        states: validateDetApiEnumList(Determinedexperimentv1State, states),
+        users: settings.user,
+      };
+      const pinnedExperimentsRequest = getExperiments(
         {
-          archived: settings.archived ? undefined : false,
-          labels: settings.label,
-          limit: settings.tableLimit,
-          name: settings.search,
-          offset: settings.tableOffset,
-          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
-          projectId: id,
-          sortBy: validateDetApiEnum(V1GetExperimentsRequestSortBy, settings.sortKey),
-          states: validateDetApiEnumList(Determinedexperimentv1State, states),
-          users: settings.user,
+          ...experimentFilter,
+          experimentFilter: { experimentIds: settings.pinned, restriction: 'RESTRICTION_INCLUDE' },
+          limit: 0,
+          offset: 0,
         },
         { signal: canceler.signal },
       );
-      setTotal(response.pagination.total ?? 0);
-      setExperiments((prev) => {
-        if (isEqual(prev, response.experiments)) return prev;
-        return response.experiments;
-      });
+      const otherExperimentsRequest = getExperiments(
+        {
+          ...experimentFilter,
+          experimentFilter: { experimentIds: settings.pinned, restriction: 'RESTRICTION_EXCLUDE' },
+          limit: settings.tableLimit - settings.pinned.length,
```

Comment:
fix: If you have more pinned experiments than your current table limit this goes negative, which doesn't have well defined behavior

Reply:
I'll add limit in pin api call